cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -308,15 +309,36 @@ private void processApplications(List<Map.Entry<ApplicationId, ApplicationSpecif
    * @throws ApplicationNotFoundException if the specified application does not exist
    */
   public ApplicationDetail getAppDetail(ApplicationId appId) throws Exception {
-    // TODO: CDAP-12473: filter based on the entity visibility in the app detail
     // user needs to pass the visibility check to get the app detail
     accessEnforcer.enforce(appId, authenticationContext.getPrincipal(), StandardPermission.GET);
-    ApplicationSpecification appSpec = store.getApplication(appId);
-    if (appSpec == null) {
+    ApplicationMeta appMeta = store.getApplicationMetadata(appId);
+    if (appMeta == null || appMeta.getSpec() == null) {
       throw new ApplicationNotFoundException(appId);
     }
     String ownerPrincipal = ownerAdmin.getOwnerPrincipal(appId);
-    return enforceApplicationDetailAccess(appId, ApplicationDetail.fromSpec(appSpec, ownerPrincipal));
+    return enforceApplicationDetailAccess(appId, ApplicationDetail.fromSpec(appMeta.getSpec(), ownerPrincipal,
+                                                                            appMeta.getDescription(),
+                                                                            appMeta.getAuthor(), appMeta.getCreated()));
+  }
+
+  /**
+   * Get details about the latest version of the specified application
+   *
+   * @param appId the id of the application to get
+   * @return detail about the latest version of the specified application
+   * @throws ApplicationNotFoundException if the specified application does not exist
+   */
+  public ApplicationDetail getLatestAppDetail(ApplicationId appId) throws Exception {
```

Comment:
Similar to the comment above, I think we take `NamespaceId nsId, String appName` as parameter instead.

Reply:
So we use appId in L332 here `accessEnforcer.enforce(appId, authenticationContext.getPrincipal(), StandardPermission.GET);`

Reply:
Looks like this method is not being used?

Reply:
this will be used in the `GET /apps/<app-id>`  api - however we are not making the api changes as part of this PR